### PR TITLE
fix: add the page.pdf capture leg, capture-only via POST /_screenshot-card

### PR DESCRIPTION
**Pull request by lukemelia**
## Summary

Second PR in the Cards to PDF chain (CS-12976); its base (#6118, the identity axes) is merged, so this diff reads clean against main.

A singular `type: 'pdf'` capture spec now produces a paged document of the settled render:

- **Engine** (`prerender/utils.ts`): a pdf branch in `captureOneEntry` runs the identical settle sequence, then emulates the media the spec asks for (`screen` today) and calls `page.pdf({ printBackground: true, preferCSSPageSize: true })`. `page.pdf()` renders `print` media by default, so pinning the media keeps the paged document on the same styles the raster path captures; the override is cleared afterward — best-effort, so a dying page surfaces its real failure rather than the reset's — so a pooled page carries no media into the next capture. Layout width is the paper's, not a viewport's: `Page.printToPDF` lays the document out at the paper's content width, where an author's `@page { size: … }` rule wins and Chrome's default paper applies otherwise. The result reports `pageCount` instead of pixel dimensions (`ScreenshotCaptureResult.width`/`height` become optional; the declared-capture path is unaffected — its own result type keeps required dims), and the response's `contentType` is derived from the capture (`application/pdf` for pdf) instead of hardcoded.
- **Bounds**, enforced post-render (the `fullPage` late-check pattern, since page count and byte size exist only after Chrome paginates): **20 pages / 10 MB** via `checkPdfCaptureBounds`, each an error naming the cap, never a truncation. Page counting scans Chromium's uncompressed `/Type /Page` dictionaries with the page-tree `/Count` as a floor (`countPdfPages`).
- **Spec rules**: pdf is **singular-only** (one `contentType` per response, so no pdf in batch entries) and refuses the raster geometry — `fullPage`, `clip`, `target`, and `viewport` (pagination lays out at paper width, so a viewport would be an inert axis minting distinct ledger identities over byte-identical documents; a viewport spelling the engine default still elides and parses) — at the shared parse, with defensive engine guards for in-process callers that assemble capture entries directly.
- **Capture-only posture** (persistence is the next PR in the chain): the GET `_screenshot/` DSL refuses `type=pdf` by name (that surface persists everything it captures and cannot yet serve paged documents), the POST handler derives no persist identity for pdf specs (`url: null`, bytes in `base64`), and the worker task's guard hashes pdf renders to null so no producer can land one in the ledger.
- **Response and CLI plumbing**: the POST response's capture entries carry `pageCount` for a paged capture (whose `width`/`height` are null — pagination has no pixel extent), and the `boxel screenshot` CLI maps `application/pdf` to a `.pdf` file extension instead of falling through to the `.png` default.

## Testing

- `media-cache-dsl-test.ts`: pdf parses on POST (singular-only, crop-mode and viewport refusals, with the default-valued viewport spelling still admitted), the GET surface refuses `type=pdf` with its own wording, `countPdfPages` unit coverage (page dictionaries, `/Count` floor, zero-marker case), `checkPdfCaptureBounds` coverage of both caps' boundary and error wording against synthesized buffers, and the task refuses to persist a pdf render under any claimed identity — all green locally.
- `screenshot-card-test.ts`: a POST pdf capture gets no persist identity and no served URL while the encoding rides the job args, and its response entry carries `pageCount` with null pixel dims — green locally.
- `boxel-cli` `screenshot.test.ts`: a pdf capture response writes its bytes to a `.pdf` file.
- `prerendering-test.ts` (real Chrome): a pdf capture of a tall card returns `%PDF-` bytes with `contentType: application/pdf` and a page count, and the next raster capture on the same pooled page still renders at the default viewport. A probe card that is short under screen media and 60000px under print pins the media: the screen-media pdf stays within the page cap, where a print render would page well past it. A 30000px-tall card errors on the page cap by name, with no truncated document riding the error.
- `runtime-common` and `realm-server` typecheck clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/realm-server/prerender/utils.ts [hunk 1]
```diff
@@ -1552,6 +1571,50 @@ async function captureOneEntry(
...
+        `capture "${entry.name}" cannot combine pdf output with target, clip, or fullPage`,
+        { title: 'Invalid screenshot capture spec' },
+      );
+    }
+    let bytes = await page.pdf({
+      printBackground: true,
+      // The author's own `@page { size: … }` rule wins; Chrome's default
+      // paper applies when the card declares none.
+      preferCSSPageSize: true,
+    });
```
Comment (lukemelia) [inline packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] **`page.pdf()` renders under print media, not the screen media this render settled under.** Verified in puppeteer-core 25.2.1: `page.pdf()` → `createPDFStream` → CDP `Page.printToPDF` with no `emulateMediaType` call, and its own docstring says so — "Generates a PDF of the page with the `print` CSS media type… To generate a PDF with the `screen` media type, call `page.emulateMediaType('screen')` before calling `page.pdf()`." The settle sequence upstream runs with no media emulation (screen), so this call re-lays-out the DOM under print media: `@media print` rules engage and `@media screen`-only content drops.

That contradicts two claims in this PR:
- the comment here ("Same settle sequence, different capture call") and the description ("a paged document of the same settled render") — the bytes are a _different_ render than what settled;
- the media-axis framing, where `media: "print"` is refused as a future milestone and `media: "screen"` is the default a pdf spec asserts — yet the pdf already renders print media.

Decision, not a mechanical fix — which media do you want?
- **Screen** (match the on-screen render): `await page.emulateMediaType('screen')` before `page.pdf()`. (`printToPDF` doesn't persist the emulation, so pooled-page hygiene needs no reset — worth confirming.)
- **Print** (the natural reading, given `preferCSSPageSize` / `printBackground`): then correct the "same settled render" comments and rethink what `media` means for pdf, so the next change doesn't build on "pdf is screen media, print is later."

The accompanying test pins neither — `a pdf capture paginates the settled render` asserts magic bytes, contentType, and pageCount, none of which separate print from screen; add a fixture with a `@media print`- or `@media screen`-only element so the chosen media is actually covered. Class: introduced here; blocking only if the answer is screen.

**Review by lukemelia — COMMENTED** [review]
[Claude Code 🤖] **Lens:** traced the pdf capture leg end to end — the engine branch and its post-render page/byte bounds, the singular-only and crop-mode refusals on both the shared parse and the engine's own guard, the `contentType` wiring from engine → render-runner → response, the optional-dims re-typing and its consumers, and the capture-only posture across the ledger identity, the GET DSL, and the worker hash guard. The base branch's `type`/`media` axes were re-checked only where this diff reads them.

**Bottom line: one thing to settle before merge — the PDF renders under _print_ media, not the _screen_ media the render settled under, which both the code comments and the media-axis design say otherwise (inline thread on `utils.ts`). It's a decision, not a mechanical fix, and blocks only if you wanted screen. Everything else lands: the bounds fail by name rather than truncate, the refusals are enforced on both the parse and the engine guard, and the capture-only wiring keeps pdf out of the ledger consistently.** Two minor non-blocking nits:

1. **Stale comment in `runtime-common/index.ts`.** On `ScreenshotPrerenderResponse`, the line "The engine produces only `image/png` today" is falsified by this PR — the engine now emits `application/pdf`. The change right above widened `ScreenshotCaptureResult` for exactly that output, so update this comment in the same pass. Evergreen nit.

2. **`pageCount` is dropped from the POST response.** In `handle-screenshot-card.ts`, a pdf capture has `spec === undefined` → `entryKey === undefined` → the capture-only normalization (`result.captures.map(...)`) runs and rebuilds each entry as name/url/width/height/deviceScaleFactor/base64. The engine computes `pageCount` and render-runner carries it on `captures[i]`, but it never reaches the response — so a pdf caller (the only surface pdf reaches today) gets the bytes without the page count already computed for the bounds check. If that's intended, ignore; if a pdf POST should expose it, add `...(c.pageCount !== undefined ? { pageCount: c.pageCount } : {})` to the map. Follow-up, not this PR.

File: packages/realm-server/prerender/utils.ts (hunk 1 above)
Comment (lukemelia) [reply packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] Resolved in 4bc828c751. Decided the `media` axis is the control here (it is the API's affordance for media type), so the pdf leg now emulates the requested media before `page.pdf()` — `screen` today, the only value the parse admits — and clears the override in `finally` for pooled-page hygiene. That makes the paged document the same screen render the raster path captures, and it carries `print` for free when a later milestone unlocks it. The "same settled render" comment is now true rather than aspirational.

Pinned by a probe card that is 200px under screen media and 60000px under print: the screen-media pdf stays within the page cap, where a print render would page well past it (`prerendering-test.ts`, real Chrome in CI). The test fails if the emulation is dropped — the print-only height trips the 20-page cap and the capture errors.

**Comment by lukemelia** [comment]
[Claude Code 🤖] Review-body nits, dispositions:

1. **Stale `index.ts` comments — fixed in 4bc828c751.** `ScreenshotPrerenderResponse.contentType` and the `type` field now say the engine produces `image/png` and `application/pdf` (and honors `pdf` as an output type), rather than "only `image/png` today".

2. **`pageCount` on the POST response — leaving it out, as a follow-up.** pdf is capture-only until the serving surfaces land, and the response shape stays as-is for now; a caller can derive the count from the bytes meanwhile. Worth revisiting when the persist/serve legs arrive.

File: packages/realm-server/prerender/utils.ts [hunk 2]
```diff
@@ -1552,6 +1571,65 @@ async function captureOneEntry(
...
+        // Pagination has no single pixel extent; the scale is the render's,
+        // reported for parity with raster captures.
+        deviceScaleFactor,
+        pageCount,
+      };
+    } finally {
+      // Restore the default (screen) media so the next capture on this pooled
+      // page renders exactly as an un-emulated one would.
+      await page.emulateMediaType();
+    }
```
Comment (habdelra) [inline packages/realm-server/prerender/utils.ts:1630]: [Claude Code 🤖] Make this restore best-effort, the way the file's other pooled-page restores are. If `page.pdf()` rejects because the target died — the case `#maybeEvict` exists for — this `await` throws too, and the capture surfaces a `Target closed` protocol error from the reset instead of the failure that actually happened. The viewport restore at the end of `captureScreenshot` and the background restore further down this function both wrap in try/catch with "Page may be closing/evicted; a best-effort restore is enough."

```suggestion
    } finally {
      // Restore the default (screen) media so the next capture on this pooled
      // page renders exactly as an un-emulated one would.
      try {
        await page.emulateMediaType();
      } catch {
        // Page may be closing/evicted; a best-effort restore is enough.
      }
    }
```

Class: introduced here. Non-blocking.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/realm-server/prerender/utils.ts [hunk 3]
```diff
@@ -1552,6 +1571,65 @@ async function captureOneEntry(
...
   output?: DeclaredEntryOutput,
 ): Promise<ScreenshotCaptureItem | RenderError> {
+  // A pdf capture paginates the settled render onto paper instead of
+  // rasterizing the viewport. Same settle sequence, different capture call —
+  // and its bounds (page count, byte size) exist only once Chrome has
+  // paginated, so they are enforced here, the fullPage late-check pattern.
+  if (entry.type === 'pdf') {
+    // Defensive: the shared parse refuses these combinations; a direct
+    // prerender-server caller could still send one.
+    if (entry.target || entry.clip || entry.fullPage) {
```
Comment (habdelra) [inline packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] The caller this names can't reach here: `/prerender-screenshot` runs the same shared parse — `parseScreenshotAttributes` in `prerender/prerender-app.ts` calls `parseScreenshotCaptureSpec` and fails the request on its error before the args reach the engine — so no HTTP caller of the prerender server can arrive with pdf + a crop mode. Same for the singular-only guard in `captureScreenshot`. The callers that do bypass the parse are the in-process ones (`captureDeclaredScreenshots`, tests).

Keep the guards, but reword so the next reader doesn't come away believing that route is unvalidated and build on it. Class: introduced here. Non-blocking.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/realm-server/prerender/utils.ts [hunk 4]
```diff
@@ -1552,6 +1571,65 @@ async function captureOneEntry(
...
+        );
+      }
+      let pageCount = countPdfPages(bytes);
+      if (pageCount > SCREENSHOT_PDF_MAX_PAGES) {
+        return buildInvalidRenderResponseError(
+          page,
+          `pdf capture "${entry.name}" produced ${pageCount} pages, over the ${SCREENSHOT_PDF_MAX_PAGES}-page cap`,
+          { title: 'PDF capture too large' },
+        );
+      }
```
Comment (habdelra) [inline packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] Neither cap has a test. `countPdfPages` is covered as a function, but nothing reaches either error: the page cap is only implicit in the media probe's premise (it asserts the *screen* render stays within the cap), and the byte cap has no coverage at all — a flipped comparison or a swapped message ships green. A screen-media fixture tall enough to page past the cap (`height: 30000px` is ~28 pages at the printed page height) pins the page-cap error the same way the probe card pins the media.

Separately, worth a decision while this block is being written: both caps are checked after Chrome has paginated and the whole buffer is in the prerender process, and unlike the `fullPage` branch below there's no pre-render extent check — a card that is genuinely tall under screen media pays the full document before the cap rejects it. A `scrollHeight` read against an implied-page-count bound would refuse it before the allocation, cheaply.

Class: follow-up (tests) / decision (pre-bound). Non-blocking.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by habdelra — COMMENTED** [review]
[Claude Code 🤖] This review went after what `page.pdf()` actually renders (checked against Chromium over CDP, driving `Page.printToPDF` with puppeteer 25.2.1's own `pdf()` defaults), the capture-only posture across every producer and consumer of the new encoding, and pooled-page hygiene. It did not run the realm-server or prerender suites.

**One blocking decision: the paged document is not the viewport render. `Page.printToPDF` re-lays the document out at the paper's content width — 816 CSS px under puppeteer's letter/zero-margin defaults — whatever viewport the capture settled at, so `viewport` is inert for pdf output and the render-parity claim doesn't hold for anything width-dependent. Detail and measurements in the reply on the media thread in `prerender/utils.ts`. Everything else is small.**

Recommendations:

1. Decide paper-from-viewport vs. keeping Chrome's paper and correcting the parity claim — reply on the media thread in `prerender/utils.ts`.
2. Make the `finally` media reset best-effort so a dead page doesn't mask the real failure — inline on that block.
3. Reword the "a direct prerender-server caller could still send one" note; that route runs the same shared parse — inline on the crop-mode guard.
4. Cover the two pdf caps, and decide whether a pre-render extent bound is worth it — inline on the cap block.
5. The one shipped client of this surface writes pdf bytes into a `.png` file: `EXTENSION_BY_CONTENT_TYPE` in `packages/boxel-cli/src/commands/screenshot.ts` maps only the three image types and falls back to `png`, while the CLI passes a `--spec` file's `captureSpec` through verbatim — so `{"card": "…", "captureSpec": {"type": "pdf"}}` lands a PDF at `<card>.png` today. One line to add the mapping. Note that touching `packages/boxel-cli/**` puts this PR under the conventional-commit title rule, so a separate follow-up may be the cleaner home. Class: introduced here (the surface newly produces `application/pdf`); non-blocking either way.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/realm-server/prerender/utils.ts (hunk 1 above)
Comment (habdelra) [reply packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] The media pinning lands — `screen` is emulated before `page.pdf()` and cleared after, and the probe card does fail if it's dropped. But the claim it's now attached to is still false, for a second reason: `page.pdf()` doesn't lay the document out at the emulated viewport at all. `Page.printToPDF` re-lays out at the paper's content width, and puppeteer's defaults are letter 8.5in with **zero** margins (`parsePDFOptions`, puppeteer-core 25.2.1) — so the pdf renders at 816 CSS px no matter what the spec's viewport says.

Measured against Chromium over CDP, screen media emulated, `printToPDF` with exactly those defaults, probe `@media (min-width: Npx) { .x { height: 20000px } }`:

| viewport | threshold ≤ 816px | threshold ≥ 820px |
| --- | --- | --- |
| 800×600 | 19 pages | 1 page |
| 1280×800 | 19 pages | 1 page |

Identical at both viewports, flipping between 816 and 820 — the layout width is the paper's, not the capture's.

Two consequences:

1. **`viewport` is inert for a pdf capture**, while remaining part of the capture identity that the persist leg will key on: two viewport-distinct pdf specs would be two ledger entries over byte-identical documents, and a caller who sets `viewport: { width: 1280, … }` to get a wide layout silently gets 816px.
2. "keeps the paged document the same render the raster path would capture" (and "a paged document of the same settled render" in the description) is still false for anything width-dependent — a `@media (min-width: 1000px)` rule the raster capture at 1280 applies, the pdf does not.

The decision, since both readings are defensible:

- **Paper follows the capture**: `page.pdf({ width: `${viewportWidth}px`, … })`, so printed layout width equals the width the render settled at and the parity claim becomes true rather than aspirational — the same move the media pinning already made.
- **Paper stays Chrome's**: then drop the render-parity claim from the comment and the description, and decide what `viewport` means on a pdf spec — refuse it alongside `fullPage`/`clip`/`target`, or document it as settle-only while staying an identity axis.

Class: introduced here. Blocking as a decision — the identity this leaves behind is what the persist and serve legs build on.

---
_Generated by [Claude Code](https://claude.ai/code)_

Comment (lukemelia) [reply packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] Resolved in d591caf75d. Decided the paper stays the print model's: a PDF is a paged document, and its width belongs to the card's `@page { size: … }` rule (which `preferCSSPageSize` honors) or Chrome's default paper — wiring paper width to a screen-geometry axis would mint non-paper page sizes and make an incidental raster knob load-bearing. What follows from that, all in the commit: `viewport` is refused on a pdf spec at the shared parse, alongside `fullPage`/`clip`/`target` (the same category — raster geometry pagination doesn't honor), so it cannot become an inert identity axis when the persist leg keys the ledger; a viewport spelling the engine default still parses, since it elides to the same canonical form as omitting it. The engine's defensive pdf guard refuses `viewport` too, and the render-parity comments and the description now say what is true: same settle and emulated media as the raster path, laid out at the paper's content width.

File: packages/realm-server/prerender/utils.ts (hunk 2 above)
Comment (lukemelia) [reply packages/realm-server/prerender/utils.ts:1630]: [Claude Code 🤖] Fixed in d591caf75d — the media reset is wrapped in the same best-effort try/catch as the file's other pooled-page restores, so a dead page surfaces the capture's real failure rather than the reset's protocol error.

File: packages/realm-server/prerender/utils.ts (hunk 3 above)
Comment (lukemelia) [reply packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] Fixed in d591caf75d. The pdf guard comments (and the crop-mode guard comment in captureScreenshot, which carried the same wording) now say the shared parse covers every wire surface — the realm-server request bodies and the prerender server's /prerender-screenshot route — and that the engine guards exist for the in-process callers that assemble capture entries directly.

File: packages/realm-server/prerender/utils.ts (hunk 4 above)
Comment (lukemelia) [reply packages/realm-server/prerender/utils.ts]: [Claude Code 🤖] Tests added in d591caf75d. The bounds check is extracted to `checkPdfCaptureBounds` (runtime-common/capture-spec.ts) and unit-covered against synthesized buffers — both caps' boundary values and exact error wording — so a flipped comparison or swapped message cannot ship green. A real-Chrome test captures a 30000px-tall card and asserts the page-cap error by name, with no truncated document riding the error. On the pre-render extent bound: declining it — a scrollHeight-based page estimate depends on the emulated media, the card's `@page` size, and `preferCSSPageSize`, so it is wrong exactly where it would matter, and the late check already caps the cost at one bounded buffer per attempt.

**Comment by lukemelia** [comment]
[Claude Code 🤖] All four review threads are addressed and resolved in d591caf75d, and the description is updated to the corrected render-parity claims plus the viewport refusal. The remaining review-body item — the CLI writing pdf bytes to a `.png` file — is fixed by 2a60e99a1c on `fix-boxel-cli-pdf-capture-extension`, opened as its own pull request since `packages/boxel-cli` changes fall under the conventional-commit title rule.

**Review by backspace — APPROVED** [review]
[Claude Code 🤖] Buck asked me to land the merge before reviewing, so this is posted from their account. The branch was conflicting against its base; I resolved and pushed the merge as `aa365a0d`. All four conflicts were unions between this branch's pdf capture-only posture and the base branch's `target` capture-only posture: the doc comment on the POST handler, the persist-identity predicate there, the same predicate in the worker task, and two test files that each added a "refuses to persist" case. Both predicates now exclude `captures`, `target`, and `type: 'pdf'`; both test cases are kept. `runtime-common` and `realm-server` typecheck clean on the result.

**Lens:** convergence on the four open threads, re-verified against Chromium rather than against the replies. **Bottom line: no blocking issues — this is ready to land.** Two one-line follow-ups below, neither worth holding for.

**Thread dispositions**

1. *Media / viewport parity* — resolved. With the media pinned, a declarative page's pdf is byte-identical across viewport widths (checked against Chromium over `Page.printToPDF`, warm-up pages discarded — the first two pages of a launch are not byte-stable), so refusing `viewport` costs nothing and the paper-width claim in `capture-spec.ts` holds as written. Worth noting the refusal's scope is right: `deviceScaleFactor`, which it deliberately leaves admitted, is *not* inert — dsf 1/2/3 produce three different documents on a page carrying a `min-resolution` query — so it stays a live axis rather than an inert one minting identities.
2. *Best-effort media reset* — resolved. The `finally` now wraps `emulateMediaType()` in try/catch like the file's other pooled-page restores.
3. *Guard comment wording* — resolved. Both guards name the shared parse as covering every wire surface and scope themselves to in-process callers.
4. *Cap coverage* — resolved. Both caps are unit-covered at the boundary and one past with exact wording, so a flipped comparison or swapped message can't ship green. I checked `countPdfPages` against real Chrome output rather than synthesized buffers: 2000px→2, 5000px→5, 21120px→20, 30000px→29 pages — so the count the cap is enforced on is accurate, and the 30000px fixture clears the 20-page cap by 9. Declining the pre-render extent bound is the right call; a `scrollHeight` estimate can't know the card's `@page` size.

**Non-blocking follow-ups**

1. **The CLI still writes pdf bytes into a `.png` file.** `EXTENSION_BY_CONTENT_TYPE` in `packages/boxel-cli/src/commands/screenshot.ts` maps only the three image types and falls back to `png`, and `--spec` passes a file entry's `captureSpec` through verbatim — so `{"card": "…", "captureSpec": {"type": "pdf"}}` lands a PDF at `<card>.png`. One line (`'application/pdf': 'pdf'`) fixes it. Class: introduced here, since the surface newly produces `application/pdf`. Touching that package puts the PR under the conventional-commit title rule, so a separate follow-up is probably the cleaner home.
2. **`pageCount` never reaches a POST caller.** The engine computes it and `render-runner` carries it on `captures[i]`, but in `handle-screenshot-card.ts` a pdf spec leaves `entryKey` undefined, so the capture-only normalization rebuilds each entry as name/url/width/height/deviceScaleFactor/base64. A pdf caller therefore gets `width: null`, `height: null` and no page count, on the one surface pdf reaches. Spreading `...(c.pageCount !== undefined ? { pageCount: c.pageCount } : {})` into that map closes it. Class: introduced here.

CI on the merge commit is still running, so this approval is on the code, not on a green run.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by backspace** [comment]
[Claude Code 🤖] Buck asked me to land the two non-blocking items from the review rather than leave them as follow-ups, so this is pushed from their account. Three commits on top of the approved head:

- **`aa365a0d`** — merge of the base branch, resolving four conflicts against the `target` capture-only work that reached it via main. Both persist-identity predicates (the POST handler and the worker task) now exclude `captures`, `target`, and `type: 'pdf'`; both new "refuses to persist" tests are kept.
- **`c8cb99d3`** — `EXTENSION_BY_CONTENT_TYPE` gains `'application/pdf': 'pdf'`, so a pdf capture no longer lands in a `.png` file. Covered by a unit test that fails without the mapping (checked by reverting it).
- **`d03663e5`** — the POST handler's capture-only normalization carries `c.pageCount` through, and `CaptureResult` gains the optional field. A pdf caller was getting `width: null`, `height: null` and no extent at all, on the one surface pdf reaches. Covered in `screenshot-card-test.ts` by an engine response the stub queue now accepts as an override.

Heads up on the title: the CLI change puts this PR under the published-package rule, so I retitled it `feat: …` — without a prefix the `boxel-cli-pr-title` check goes red, and `feat` is the right bump for a new output type the CLI can now file correctly. If you would rather keep this PR server-only, dropping `c8cb99d3` and restoring the plain title puts it back.

`runtime-common`, `realm-server` and `boxel-cli` typecheck clean; eslint and prettier clean on all four files; the boxel-cli suite passes locally (18/18). The realm-server suite is left to CI.
